micro-rdk/src/common/board.rs

```diff
@@ -102,20 +103,20 @@ pub trait Board: Status + DoCommand {
 }
 
 /// An alias for a thread-safe handle to a struct that implements the [Board] trait
-pub type BoardType = Arc<Mutex<dyn Board>>;
+pub type BoardType = Arc<Mutex<dyn Board<u16>>>;
```

Comment:
i don't like the idea to tie a generic to Board especially that for now we only read `u16` Did you try W=u16? otherwise I would leave it to u16 default for now we can revisit it once we need to support another integer type

Reply:
I'm not sure I understand what you mean by W=u16. I did try this
```rust
pub type BoardType = Arc<Mutex<dyn Board<AnalogReaderWord = u16>>>
```
but it did not compile